packages/core/src/Site/index.js

```diff
@@ -1300,11 +1303,67 @@ class Site {
 
           return publish(basePath, options);
         })
-        .then(resolve)
+        .then(() => {
+          const git = simpleGit({ baseDir: process.cwd() });
+          const options = {};
+          options.remote = defaultDeployConfig.remote;
+          options.repo = this.siteConfig.deploy.repo || defaultDeployConfig.repo;
+          options.branch = this.siteConfig.deploy.branch || defaultDeployConfig.branch;
+          return Site.getDeploymentUrl(git, options);
+        })
+        .then(depUrl => (depUrl != null ? resolve(`Deployed at ${depUrl}!`) : resolve('Deployed!')))
         .catch(reject);
     });
   }
 
+  /**
+   * Gets the url where the website is deployed at.
+   */
+  static getDeploymentUrl(git, options) {
+    const HTTPS_PART = 'https://';
+    const SSH_PART = 'git@github.com:';
+    const GITHUB_IO_PART = 'github.io';
+
+    // https://<name|org name>.github.io/<repo name>/
+    function constructGhPagesUrl(remoteUrl) {
+      if (remoteUrl.includes(HTTPS_PART)) {
```

Comment:
should we use `startsWith` here instead?

Likewise for `SSH_PART` (i'm not too sure on the format requirements here though, correct me if I'm wrong!)